source/include/jobs.h

```diff
@@ -111,11 +111,111 @@
+#define JOBS_API_LEVEL_SEPARATOR          "/"
+
 #define JOBS_API_COMMON_LENGTH( thingNameLength ) \
     ( JOBS_API_PREFIX_LENGTH + ( thingNameLength ) + JOBS_API_BRIDGE_LENGTH )
 
 /** @endcond */
 
+
+/**
+ * @cond DOXYGEN_IGNORE
+ * Doxygen should ignore this macro as it is private.
+ */
+
+/* AWS IoT Jobs API topics. */
+#define JOBS_TOPIC_COMMON( thingName, jobId, jobsApi ) \
+    ( JOBS_API_PREFIX                                  \
+      thingName                                        \
+      JOBS_API_BRIDGE                                  \
+      jobId                                            \
+      jobsApi )
+/** @endcond */
+
+/**
+ * @ingroup jobs_constants
+ * @brief Topic string for subscribing to the NextJobExecutionChanged API.
+ *
+ * This macro should be used when the thing name is known at the compile time.
+ * If the thing name is not known at compile time, the #Jobs_GetTopic API
+ * should be used instead.
+ *
+ * @param thingName The thing name as registered with AWS IoT Core.
+ */
+#define JOBS_API_SUBSCRIBE_NEXTJOBCHANGED( thingName ) \
+    JOBS_TOPIC_COMMON( thingName, JOBS_API_JOBID_NULL, JOBS_API_NEXTJOBCHANGED )
+
+/**
+ * @ingroup jobs_constants
+ * @brief Topic string for subscribing to the JobExecutionsChanged API.
+ *
+ * This macro should be used when the thing name is known at the compile time.
+ * If the thing name is not known at compile time, the #Jobs_GetTopic API
+ * should be used instead.
+ *
+ * @param thingName The thing name as registered with AWS IoT Core.
+ */
+#define JOBS_API_SUBSCRIBE_JOBSCHANGED( thingName ) \
+    JOBS_TOPIC_COMMON( thingName, JOBS_API_JOBID_NULL, JOBS_API_JOBSCHANGED )
+
+/**
+ * @ingroup jobs_constants
+ * @brief Topic string for publishing to the StartNextPendingJobExecution API.
+ *
+ * This macro should be used when the thing name is known at the compile time.
+ * If the thing name is not known at compile time, the #Jobs_StartNext API
+ * should be used instead.
+ *
+ * @param thingName The thing name as registered with AWS IoT Core.
+ */
+#define JOBS_API_PUBLISH_STARTNEXT( thingName ) \
+    JOBS_TOPIC_COMMON( thingName, JOBS_API_JOBID_NULL, JOBS_API_STARTNEXT )
+
+/**
+ * @ingroup jobs_constants
+ * @brief Topic string for publishing to the GetPendingJobExecutions API.
+ *
+ * This macro should be used when the thing name is known at the compile time.
+ * If the thing name is not known at compile time, the #Jobs_GetPending API
+ * should be used instead.
+ *
+ * @param thingName The thing name as registered with AWS IoT Core.
+ */
+#define JOBS_API_PUBLISH_GETPENDING( thingName ) \
+    JOBS_TOPIC_COMMON( thingName, JOBS_API_JOBID_NULL, JOBS_API_GETPENDING )
+
+/**
+ * @ingroup jobs_constants
+ * @brief Topic string for publishing to the DescribeJobExecution API.
+ *
+ * This macro should be used when the thing name and jobID are known at the
+ * compile time. If next pending job is being queried, use $next as job ID.
```

Comment:
Is job ID ever known at compile time?

Reply:
This macro's main utility would be for using the `$next` job ID. If you think that we should provide macros only for that case, I can limit the macro to generate topics only for the next job ID.